### ISSUE TITLE
XCTest extensions

### ISSUE THREAD

**Issue by BennX** [SwifterSwift/SwifterSwift#495]
I'd like to see/add extensions which are useful for testing. 

One of my most used ones is an `AssertElementExists` method as `XCTestCase` extension for UITesting (If you don't use Snapshot testing). It does wait for an `XCUIElement` to appear and taps it if wanted. If the element doesn't appear it'll record an error in the line the line where you used the `AssertElementExists` using default parameters which get filed with the `#file` `#line`.

I have one thing that concerns me and that is that it requires the target to include XCUnit framework which normally should only exist in test targets. It may makes sense to create a whole new target for this or shouldn't SwifterSwift contain test extensions?

What's your opinion on it?

**Comment by SD10** [comment]
This is an interesting proposal 🤔 My 2 cents are: we probably shouldn't try to get involved with `XCTest` because we'll never compete with a dedicated library with complex matchers like `Quick/Nimble`.

**Comment by BennX** [comment]
@SD10 I agree on that. In my opinion SwifterSwift shouldn't compete with them but there might be tiny useful extensions like the `AssertElementExists` for ui testing.

**Comment by marcocapano** [comment]
It is also true that SwifterSwift extension are made so that you can copy and import them individually without importing the whole library, which means this could be useful for folks that just want a little helper method without using nimble or other libraries
